### ISSUE TITLE
Update and Create method inference

### ISSUE THREAD

**Issue by sayre1000** [DannyBen/pundit_extra#17]
So in CanCanCan:

> If you specify a `create_params` or `update_params` method, CanCan will run that method depending on the action you are performing.

This same thing does not currently work with the punidt_extra gem. I have made my own version of the gem that supports this functionality by altering the `resource_attributes` method from this:

```
def resource_attributes(resource, action)
      if has_permitted_attributes? resource, action
        permitted_attributes(resource)
      else
        send :"#{resource_name}_params"
      end
    end
```
to:

```
def resource_attributes(resource, action)
      if has_permitted_attributes?(resource, action)
        permitted_attributes(resource)
      else
        params_method = "#{action}_params"
        if respond_to?(params_method, true)
          send(params_method)
        else
          resource_params_method = "#{resource_name}_params"
          if respond_to?(resource_params_method, true)
            send(resource_params_method)
          else
            {}
          end
        end
      end
    end
 ```
 
 I have confirmed this works by using my modified version of the gem in the CanCanCan project I am porting over but was unable to figure out a way to meaningfully test it. I was also unable to create a branch due to permission restrictions.

**Comment by sayre1000** [comment]
Forgot to add that I also added this:

```
      elsif action == 'update'
        resource = scope.find params[:id]
        resource.attributes = resource_attributes(resource, action)
```

**Comment by DannyBen** [comment]
I will take a look into adding it, thanks for pointing that out.

**Comment by DannyBen** [comment]
I have implemented this with a slightly modified version of the code you provided, and added a spec for it.

Since it has been some time since I last used Rails - can you confirm the PR?

**Comment by DannyBen** [comment]
Released in version 0.3.1 - thanks for the ticket and participation.
